pysiaf/utils/tools2.py

```diff
@@ -0,0 +1,217 @@
+"""A collection of helper functions to support pysiaf calculations and functionalities
+
+Authors
+-------
+
+    - Colin Cox
+
+References
+----------
+
+
+
+"""
+
+import sys
+import numpy as np
+from math import *
+import copy
+import pylab as pl
+
+from astropy.table import Table
+
+from ..aperture import PRD_REQUIRED_ATTRIBUTES_ORDERED
+from .polynomial import poly, ShiftCoeffs, Rotate, FlipY, FlipX, rotate_coefficients, RotateCoeffs, triangle, invert
+from ..siaf import Siaf
+from pysiaf.aperture import DISTORTION_ATTRIBUTES
+
+def checkinv(A, B, C, D, order):
+    '''Check that forward and inverse transformations are consistent'''
+    [x,y] = np.mgrid[-200:201:200, -200:201:200]
+    v2 = poly(A, x, y, order)
+    v3 = poly(B, x, y, order)
+    x2 = poly(C, v2, v3, order)
+    y2 = poly(D, v2, v3, order)
+
+    print ('     x         y          V2        V3        xp        yp       dx        dy')
+    ######   -200.0000 -200.0000  -12.5849  -12.5939 -199.7371 -200.2741    0.2629   -0.2741
+
+    for i in range(3):
+        for j in range(3):
+            print(8*'%10.4f' %(x[i,j],y[i,j],v2[i,j],v3[i,j],x2[i,j],y2[i,j], x2[i,j]-x[i,j], y2[i,j]-y[i,j]))
+
+    # Summary
+    print ('\nShifts', (x2-x).mean(), (y2-y).mean())
+    print ('RMS {:10.2e} {:10.2e}'.format( (x2-x).std(), (y2-y).std() ))
+    print()
+
+    return
+
+def get_polynomial_coefficients(aperture, group):
```

Comment:
The aperture class now provides the `get_polynomial_coefficients` method. Please use that instead.